packages/python/pyfora/src/core/PyObjectPtr.hpp

```diff
@@ -0,0 +1,232 @@
+/***************************************************************************
+   Copyright 2016 Ufora Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+****************************************************************************/
+#pragma once
+
+#include <Python.h>
+
+#include <cstddef>
+
+
+class PyObjectPtr {
+public:
+    PyObjectPtr(PyObject* pyObject)
+        : mPtr(pyObject)
+        {
+        }
+
+    PyObjectPtr()
+        : mPtr(nullptr)
+        {
+        }
+
+    static PyObjectPtr Incremented(PyObject* pyObject)
```

Comment:
The common use case I had was wrapping a "new reference" to a `PyObject` (meaning already increfed). For the most part, python C-api functions return such references (but not always). For example, PyObject_Call(...) returns a new reference to its result, and there's no need to incref.